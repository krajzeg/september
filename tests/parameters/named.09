testTriangle := |a,b,c=5| {
	if (a*a + b*b == c*c) {
		print(a, b, c, "specifies a right triangle.")
	} else {
		print(a, b, c, "does not specify a right triangle.")
	}
}

print("Trying a jumbled up order.")
testTriangle(c:5, a:3, b:4)
print("Mixing positional, named and default arguments.")
testTriangle(b:4, 3)

print("Duplicate arguments should cause exceptions (two tests).")
try {
	testTriangle(5, 4, a:3)
	print("But they don't.")
} catch (EWrongArguments) {
	print("And they do.");
}
try {
	testTriangle(a:5, a:4, a:3)
	print("But they don't if you use only named args.")
} catch (EWrongArguments) {
	print("And they do again.");
}
