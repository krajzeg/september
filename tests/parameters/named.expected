Trying a jumbled up order.
3 4 5 specifies a right triangle.
Mixing positional, named and default arguments.
3 4 5 specifies a right triangle.
Duplicate arguments should cause exceptions (two tests).
And they do.
And they do again.
